### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by ghermaico135**

### CODE REVIEW

**Review by Favourezeugwa — COMMENTED** [review]
Hi team 👋🏽,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ❌ ❌❌❌

You have submitted a project, that does not include a descriptive Pull request summary.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by ghermaico135** [comment]
Preserve Code to localstorage-is after submitting the data it stored on the web browser lcoalstorage.So this code set and get items to localstorage on the browser.

File: js/preserveData.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+// set userDate to localStorage
+const setUserData = { name: '', email: '', msg: '' };
+
+submissionBtn.addEventListener('click', () => {
+  setUserData.name = userName.value;
+  setUserData.email = userEmail.value;
+  setUserData.msg = userMessage.value;
+  localStorage.setItem('setUserData', JSON.stringify(setUserData));
+});
```
Comment (LionRouge1) [inline js/preserveData.js]: - [ ] Great job preserving the data in the browser :clap: :sunglasses: However data should be saved in the browser when the user changes the content of any input field. But you're saving the data when the user clicks on the button submit, Kindly check it

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @ghermaico135, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR :heavy_check_mark: 
- Good commit :+1: 
- All linters passed :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ghermaico135** [comment]
Changing the event handler to save on submit rather than on click

File: js/preserveData.js (hunk 1 above)
Comment (divinecharlotte) [inline js/preserveData.js]: - [ ] As requested by the previous reviewer kindly consider hitting the below  points as they are requirements 
-  When the user changes the content of any input field, the data is saved to the local storage.
- When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.

 ☠️  Do not create one local storage entry per input field. just create one local storage and as long as the user updates one field the whole local storage gets updated


doing the above will make your project stand out and fulfill all the requirements

**Review by divinecharlotte — CHANGES_REQUESTED** [review]
Hi @Team , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive PR ✔️ 

## Required Changes ♻️

_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/preserveData.js (hunk 1 above)
Comment (joel-kalema) [inline js/preserveData.js]: - [ ] Good job @ghermaico135 😃 , as the [previous reviewer](https://github.com/ghermaico135/Portfolio/pull/13#:~:text=However%20data%20should%20be%20saved%20in%20the%20browser%20when%20the%20user%20changes%20the%20content%20of%20any%20input%20field.%20But%20you%27re%20saving%20the%20data%20when%20the%20user%20clicks%20on%20the%20button%20submit%2C%20Kindly%20check%20it) said,  data should be saved in the browser when the user changes the content of any input field. But you're saving the data when the user clicks on the button submit, which means that the data will be saved when the user will be changing the content of any input field, not only when he clicks on the button submit. Kindly check it 👍

**Review by joel-kalema — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi Team
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me @joel-kalema  when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ghermaico135** [comment]
I Added "change eventlistner" which captures the change on localstorage when users input to the field automatically

**Review by amejid — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Linter checks passing ✔️ 
- Correct use of Github flow ✔️ 
- Data getting preserved in local storage ✔️ 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by joel-kalema** [comment]
> I Added "change eventlistner" which captures the change on localstorage when users input to the field automatically

Hello @ghermaico135, this is what we have  😃, if we refresh the page, data will not be saved  

Capture

**Pull request merged**